wger/weight/views.py

```diff
@@ -150,9 +150,29 @@ def overview(request, username=None):
                                      'month': max_date.month,
                                      'day': max_date.day}
 
+    last_five_weight_entries = WeightEntry.objects.filter(user=user).order_by('-date')[:5]
+    last_five_weight_entries_details = []
+
+    if last_five_weight_entries:
+        i = 0 
+
+        while (i != len(last_five_weight_entries)-1):
+            curr_entry = last_five_weight_entries[i]
+            prev_entry = last_five_weight_entries[i+1]
+            weight_diff = curr_entry.weight - prev_entry.weight 
+            if weight_diff > 0:
+                weight_diff = '+'+str(weight_diff)
+            day_diff = (curr_entry.date - prev_entry.date).days
+            last_five_weight_entries_details.append((curr_entry, weight_diff, day_diff))
+            i += 1
+
+        curr_entry = last_five_weight_entries[i] 
+        last_five_weight_entries_details.append((curr_entry,'-/-', '-/-'))
+
```

Comment:
While the code works, the loop doesn't feel *pythonic*. I would suggest doing something along these lines (not tested!):

```python
curr_entry = None
for entry in last_five_weight_entries:
    prev_entry = curr_entry
    curr_entry = entry
    if prev_entry and curr_entry:
        weight_diff = curr_entry.weight - prev_entry.weight
        day_diff = (curr_entry.date - prev_entry.date).days
        # and so on...
```